dapr_agents/agents/durable.py

```diff
@@ -236,35 +346,122 @@ def agent_workflow(self, ctx: wf.DaprWorkflowContext, message: dict):
                         )
                         for idx, tc in enumerate(tool_calls)
                     ]
-                    yield ctx.when_all(parallel)
-                    task = None  # prepare for next turn
+                    tool_results = yield wf.when_all(parallel)
+
+                    handoff_results = [
+                        result
+                        for result in tool_results
+                        if result and result.get("is_handoff")
+                    ]
+                    persisted_results = [
+                        result
+                        for result in tool_results
+                        if result and not result.get("is_handoff")
+                    ]
+
+                    if persisted_results:
+                        yield ctx.call_activity(
+                            self.persist_tool_results,
+                            input={
+                                "instance_id": ctx.instance_id,
+                                "tool_results": persisted_results,
+                                "time": ctx.current_utc_datetime.isoformat(),
+                            },
+                        )
+
+                    if handoff_results:
```

Comment:
how is handoff different than supporting agents as tools? or is this basically just that?